src/ir/ssa_names.h

```diff
@@ -17,51 +17,89 @@
 #define SRC_IR_SSA_NAMES_H_
 
 #include "absl/container/flat_hash_map.h"
+#include "src/ir/value.h"
 
 namespace raksha::ir {
 
 class Operation;
 class Block;
 class Module;
+class Value;
 
 class SsaNames {
  public:
-  using ID = int;
-  SsaNames() {}
-  // Disable copy (and move) semantics.
+  using ID = std::string;
+  SsaNames()
+      : block_ids_("b"),
+        module_ids_("m"),
+        value_ids_("%"),
+        operation_ids_("%") {}
+
+  // Disable copy semantics.
   SsaNames(const SsaNames &) = delete;
   SsaNames &operator=(const SsaNames &) = delete;
+  SsaNames(SsaNames &&) = default;
 
   ID GetOrCreateID(const Operation &operation) {
     return operation_ids_.GetOrCreateID(&operation);
   }
 
+  ID UpdateID(const Operation &operation, ID id) {
```

Comment:
As we talked about at lunch, let's make this `AddID` and have it `CHECK`-fail if there is already an entry for the item being added.